lib/Ethereum/RPC/Contract.pm

```diff
@@ -80,27 +85,28 @@ it to the contract class.
 
 =item gas_price => numeric (optional)
 
+=item max_fee_per_gas => numeric (optional)
+
+=item max_priority_fee_per_gas => numeric (optional)
+
 =back
 
 =cut
 
 sub BUILD {
     my ($self) = @_;
-    my @decoded_json = @{decode_json($self->contract_abi // "[]")};
+    my @decoded_json = @{ decode_json($self->contract_abi // "[]") };
 
     for my $json_input (@decoded_json) {
         if ($json_input->{type} =~ /^function|event|constructor$/) {
-            push(@{$self->contract_decoded->{$json_input->{name} // $json_input->{type}}}, $json_input->{inputs}) if scalar @{$json_input->{inputs}} > 0;
+            push(@{ $self->contract_decoded->{ $json_input->{name} // $json_input->{type} } }, $json_input->{inputs});
```

Comment:
any particular reason we don't need if condition anymore?

Reply:
yes, because it will ignore function without args for sample `decimals()` and we want to include these.